backend/middlewares/sessionManager.js

```diff
@@ -28,6 +30,11 @@ module.exports = async (req, res, next) => {
 	}
 
 	session(req, res, function(err) {
+		if (req.headers.referer && req.session.user && req.session.user.referer && !req.headers.referer.match(req.session.user.referer)) {
```

Comment:
should this check just go here? https://github.com/3drepo/3drepo.io/blob/master/backend/middlewares/checkPermissions.js#L74